src/L1_headers_store/contract.cairo

```diff
@@ -54,15 +55,109 @@ mod L1HeaderStore {
             block_header_rlp_bytes_len: usize,
             block_header_bytes: Array<usize>,
             block_header_rlp: Array<u64>,
-        ) {}
+        ) {
+            let child_block_parent_hash = self.block_parent_hash.read(block_number + 1);
+
+            validate_provided_header_rlp(
+                child_block_parent_hash, block_number, block_header_rlp_bytes_len, block_header_rlp
+            );
+
+            let rlp = IntsSequence(
+                block_header_rlp, block_header_rlp.length, block_header_rlp_bytes_len
+            );
+            let parent_hash = decode_parent_hash(rlp);
+            self.block_parent_hash.write(block_number, parent_hash);
+
+            let mut i = 0;
+            loop {
```

Comment:
No need for loop, you need to use the option: ProcessBlockOptions enum to decide which data from the block you need to extract